src/units/stimuli.sequence.dragdrop.pas

```diff
@@ -169,29 +177,57 @@ procedure TDragDropStimuli.RightDragDrop(Sender, Source: TObject; X, Y: Integer
 var
   Sample : TDragDropableItem;
   Comparison : TDragDropableItem;
+  LAnimation : TAnimation;
+  FDragDropDone : Boolean = False;
 begin
   Sample := Source as TDragDropableItem;
   Comparison := Sender as TDragDropableItem;
 
   Sample.Color := clGreen;
   Sample.Left := Comparison.Left;
   Sample.Top := Comparison.Top - Sample.Height - 10;
-  FAnimation.Join(Comparison);
+
+  LAnimation := TAnimation.Create(Parent);
+  LAnimation.Parent := Parent;
+  LAnimation.Join(Sample, Comparison);
+  LAnimation.SendToBack;
+  LAnimation.Show;
+  FDoneAnimations.Add(LAnimation);
+
+  Sample.Draggable:=False;
+
+  for Sample in FSamples do
+    if Sample.Draggable then begin
+      FDragDropDone := False;
+      FAnimation.Animate(Sample);
+      Break;
+    end else begin
+      FDragDropDone := True;
+    end;
+  if FDragDropDone then begin
+    FAnimation.Stop;
+    FAnimation.Hide;
+  end;
 end;
 
 procedure TDragDropStimuli.SetFocus(Sender: TObject; Button: TMouseButton;
   Shift: TShiftState; X, Y: Integer);
+var
+  LSample : TDragDropableItem;
 begin
-  FAnimation.Animate(Sender as TLightImage);
+  LSample := Sender as TDragDropableItem;
+  if LSample.Draggable then
+    if FAnimation.Sibling <> (LSample as TLightImage) then
+      FAnimation.Animate(LSample);
```

Comment:
O foco é atribuido apenas se o estímulo móvel ainda for arrastável (antes de ser fixado em uma resposta correta) e apenas se for diferente dele mesmo (neste caso o `Sender`).

Reply:
Ótimo professor!